elasticgraph-schema_definition/lib/elastic_graph/schema_definition/factory.rb

```diff
@@ -154,6 +154,20 @@ def new_filter_input_type(source_type, name_prefix: source_type, category: :filt
             EOS
           end
 
+          t.field @state.schema_elements.all_of, "[#{t.name}!]" do |f|
+            f.documentation <<~EOS
+              Matches records where all of the provided sub-filters evaluate to true.
+              This works just like an AND operator in SQL.
+
+              Note: multiple filters are automatically ANDed together. This is only needed when you have multiple filters that can't
+              be provided on a single `#{t.name}` input because of collisions between key names. For example, if you want to provide
+              multiple `#{any_satisfy}: ...` filters, you could do `#{all_of}: [{#{any_satisfy}: ...}, {#{any_satisfy}: ...}]`.
+
+              Will be ignored when `null` is passed. 
+              When an empty list is passed, this part of the filter matches no documents.
```

Comment:
@ayousufi please retain the documentation string we had before. In particular, `Will be ignored when null is passed` is misleading (we don't strictly ignore empty filters anymore...) and we went through and updated all the docs to provide more accurate wording.